src/coreclr/vm/interpexec.cpp

```diff
@@ -21,6 +21,12 @@ InterpThreadContext* InterpGetThreadContext()
         // FIXME VirtualAlloc/mmap with INTERP_STACK_ALIGNMENT alignment
         threadContext->pStackStart = threadContext->pStackPointer = (int8_t*)malloc(INTERP_STACK_SIZE);
         threadContext->pStackEnd = threadContext->pStackStart + INTERP_STACK_SIZE;
+        threadContext->pFrameDataAllocator = new FrameDataAllocator(INTERP_STACK_FRAGMENT_SIZE);
```

Comment:
It would be better to allocate `INTERP_STACK_FRAGMENT_SIZE` chunk only once there is actual stackalloc. We try to do things lazily where possible since it is good for performance.

Once you do that, `FrameDataAllocator` can be inline field of `InterpThreadContext`.

Reply:
The FrameDataAllocator is now inlined and InterpThreadContext instance is moved into the CoreCLR Thread class for correct destructor invocation. InterpThreadContext is created lazily on the first `pThread->GetInterpThreadContext()` in prestub.cpp, while FrameDataAllocator allocates the first block on the first localloc.